planner/core/preprocess.go

```diff
@@ -835,7 +835,7 @@ func (p *preprocessor) checkDropSequenceGrammar(stmt *ast.DropSequenceStmt) {
 func (p *preprocessor) checkDropTableGrammar(stmt *ast.DropTableStmt) {
 	p.checkDropTableNames(stmt.Tables)
 	enableNoopFuncs := p.ctx.GetSessionVars().EnableNoopFuncs
-	if stmt.TemporaryKeyword == ast.TemporaryLocal && !enableNoopFuncs {
+	if stmt.TemporaryKeyword != ast.TemporaryNone && !enableNoopFuncs {
```

Comment:
No, dropping global temporary tables is fine for `DROP GLOBAL TEMPORARY TABLE ...`.
E.g.
```sql
create global temporary table temp(id int) on commit delete rows;
drop global temporary table temp;   # fine
```